File: BinText.java
/*
created by Ahsan
*/
import java.io.*;
import java.nio.ByteBuffer;
import java.util.*;

public class BinText{
	 
	public static ArrayList <Byte> a = new ArrayList<Byte>();
	public static ArrayList <Byte> byteFile = new ArrayList<Byte>();
	public static void main(String args[]) throws IOException{
		DataInputStream  input = new DataInputStream(new FileInputStream(System.getProperty("user.dir")+"/LOG.txt"));
		while(input.available() > 0){
			//System.out.println("loop er moddhe");
			byte b = input.readByte();
			a.add(b);
			//String x= String.format("%02x", b & 0xFF);
			//System.out.println(x);
		}
		input.close();
		
		int n= a.size();
		byte[] bytearray = new byte[a.size()];
		int idx=0;
		for (Byte b:a)bytearray[idx++]=b.byteValue();
		for(int i = 0; i < n; i+=22){
			byte[] deviceID = new byte[4];
			new Random().nextBytes(deviceID);
			for (byte d:deviceID) byteFile.add(d);
			for(int j = 0; j < 22; j++){
				byteFile.add(a.get(i+j));
			}
			
			int year = a.get(i).intValue();
			int month = a.get(i+1).intValue();
			int date = a.get(i+2).intValue();
			int hour = a.get(i+3).intValue();	
			int minute = a.get(i+4).intValue();
			int second = a.get(i+5).intValue();	
			int cf = a.get(i+6).intValue();
			ByteBuffer wrapper = ByteBuffer.wrap(bytearray, i+7, 4);
			int cfid = wrapper.getInt();
			
			
			System.out.println(year+ " "+ month+" "+date+" "+hour+" "+minute+" "+second+ " "+cf+ " "+ cfid);
		}
		String fileName = System.getProperty("user.dir")+"/outLog.txt";
				
		FileWriter writer = new FileWriter(fileName); 
		for(byte b: byteFile) {
		  writer.write(b);
		}
		writer.write(0xff);
		writer.close();
	}

}

File: bintext.py
from __future__ import print_function
import random
import os
import struct
import MySQLdb
import glob


def disconnectdb(db):
    db.commit()
    db.close()
    return



def query(cursor):

    #result_cnt = cursor.execute("select `sys_id` from `syscall_map` where `syscall`= '"+s+"'")
    #if(result_cnt==0):
        #print(s)
    sql = "INSERT INTO `log` (`deviceID`, `year`, `month`, `date`, `hour`, `minute`, `second`, `cf`, `cfID`) VALUES ('"+ str(int(devID,16)) +"','"+ str(year) + "','" + str(month) + "','"+ str(date) + "','"+ str(hour) + "','"+ str(minute)+ "','"+ str(second) + "','"+ str(cf) + "','"+ str(cfID) + "');"
    print(sql)
    cursor.execute(sql)
    #cursor.execute("select `sys_id` from `syscall_map` where `syscall`= '"+s+"'")

##    result = cursor.fetchone()
##    sys_id=result[0]
##    syscall_count[sys_id] += 1
##
##    f2.write(str(sys_id)+"\n")
##    prev_sys_id = sys_id
##
##    cursor.execute("SELECT max(`sys_id`) FROM `syscall_map`;");
##    mapped_so_far=  cursor.fetchone()[0]


def database_work():

    db = MySQLdb.connect(host="localhost", port=3306, user="root", passwd="", db="fingerprint")
    cursor = db.cursor()
    query(cursor)
    disconnectdb(db)


piece_size= 22
with open("outLog.txt", "wb") as outFile:
    with open("LOG.txt", "rb") as inFile:
        while True:
            piece = inFile.read(piece_size)

            if piece == "":
                break # end of file
            deviceID = os.urandom(4)
            outFile.write(deviceID)
            outFile.write(piece)
            devID = deviceID.encode('hex')
            print('device ID: %s = ' %devID + str(int(devID,16)))

            
            log = piece.encode('hex')
            year = int(log[0:2], 16)
            month = int(log[2:4], 16)
            date = int(log[4:6], 16)
            hour = int(log[6:8], 16)
            minute = int(log[8:10], 16)
            second = int(log[10:12], 16)
            cf = int(log[12:14], 16)
            cfID = int(log[14:], 16)
            print(log);
            print("%d %d %d %d %d %d %d %s" %(year,month,date,hour,minute,second,cf,cfID))

            database_work()            
            
        outFile.write(struct.pack('1B', 0xff))


